packages/common/infra/src/modules/doc/stores/docs.ts

```diff
@@ -59,20 +59,31 @@ export class DocsStore extends Store {
   }
 
   watchDocMeta(id: string) {
+    let docMetaIndexCache = -1;
     return yjsObserveByPath(
       this.workspaceService.workspace.rootYDoc.getMap('meta'),
       'pages'
     ).pipe(
       switchMap(yjsObserve),
       map(meta => {
         if (meta instanceof YArray) {
-          let docMetaYMap = null as YMap<any> | null;
-          meta.forEach(doc => {
-            if (doc.get('id') === id) {
-              docMetaYMap = doc;
+          if (docMetaIndexCache >= 0) {
```

Comment:
will the index change when any page is permanently removed from workspace meta?

Reply:
After cache, there is also a ID check